src/lib.rs

```diff
@@ -14,15 +14,33 @@ use dprint_core::configuration::GlobalConfiguration;
 use dprint_core::generate_plugin_code;
 use dprint_core::plugins::SyncPluginHandler;
 
+use schemars::{schema_for, JsonSchema};
+
 #[derive(Clone, Serialize)]
 #[serde(rename_all = "camelCase")]
+#[derive(JsonSchema)]
+// Adjust names with dprint global configuration
+//
+// List of upstream:
+//   dprint: https://github.com/dprint/dprint/blob/0.49.0/crates/core/src/configuration.rs#L257-L278
+//   typestyle: https://github.com/Enter-tainer/typstyle/blob/v0.12.14/crates/typstyle-core/src/config.rs#L4-L11
+// TODO: Remove required from all options in json schema. See https://github.com/GREsau/schemars/issues/344
```

Comment:
dropping schemars might be better here. Other plugins looks not using these crate and it will not fit for dynamic use of `get_value`